src/openrct2/CmdlineSprite.cpp

```diff
@@ -452,6 +457,157 @@ int32_t cmdline_for_sprite(const char** argv, int32_t argc)
         sprite_file_close();
         return 1;
     }
+    else if (_strcmpi(argv[0], "exportalldat") == 0)
+    {
+        if (argc < 3)
+        {
+            fprintf(stdout, "usage: sprite exportall <spritefile> <output directory>\n");
+            return -1;
+        }
+
+        char datName[9] = { 0 };
+        std::fill_n(datName, 8, ' ');
+        int32_t i = 0;
+        for (const char* ch = argv[1]; *ch != '\0' && i < 8; ch++)
+        {
+            datName[i++] = *ch;
+        }
```

Comment:
Would doing `safe_strcpy(outputPath, argv[2], MAX_PATH);` like the other place work?

Reply:
No. This concerns a .DAT name, consisting of 8 characters, not a full path.